### PR TITLE
Fix clang-format version parsing

### PR DESCRIPTION
**Pull request by Luthaf**
Ubuntu provides alternatives version with a different version string (Ubuntu clang-format version 11.0.0-2~ubuntu20.04.1)

Also downgrade the fatal error if the version do not match to a status message and disable `pretty-cpp`. This way users of the code have an easier time installing it.

----

Reported by @rosecers after the code failed to build in CI (https://github.com/cosmo-epfl/kernel-tutorials/runs/2711166252).

### CODE REVIEW

**Comment by max-veit** [comment]
Good catch, thanks!

Perhaps not entirely related to this PR, but there have been complaints that `make pretty-cpp` just appears to not be a valid make target if `clang-format` was not detected at configure time, resulting in some confusion (was this a typo? did the make target stop working?). If it's an easy fix -- something like just having `make pretty-cpp` output a useful error, like "please make sure `clang-format` is available when you run cmake`" -- then it would be nice to quickly shoehorn it into this PR.

**Comment by Luthaf** [comment]
> Perhaps not entirely related to this PR, but there have been complaints that make pretty-cpp just appears to not be a valid make target if clang-format was not detected at configure time, resulting in some confusion (was this a typo? did the make target stop working?). If it's an easy fix -- something like just having make pretty-cpp output a useful error, like "please make sure clang-format is available when you run cmake`" -- then it would be nice to quickly shoehorn it into this PR.

Done! The `pretty-cpp` target is always created, and a message is sent both at configure and build time if clang-format can not be found

```
$ make pretty-cpp
***ERROR*** clang-format not found, the pretty-cpp target have been disabled
make[3]: *** [CMakeFiles/clang-format-not-found] Error 1
make[2]: *** [CMakeFiles/clang-format-not-found.dir/all] Error 2
make[1]: *** [CMakeFiles/pretty-cpp.dir/rule] Error 2
make: *** [pretty-cpp] Error 2
```

**Comment by Luthaf** [comment]
Would you have time to give this a second round of review @max-veit?

File: cmake/clangformat.cmake [hunk 1]
```diff
@@ -62,24 +64,48 @@ mark_as_advanced(CLANG_FORMAT)
...
 if(CLANG_FORMAT)
     message(STATUS "clang-format executable: ${CLANG_FORMAT}")
-    # common target to concatenate all cpplint.py targets
-    add_custom_target(${CLANG_FORMAT_TARGET})
     set(CLANG_FORMAT_FOUND TRUE)
 else()
-  message(STATUS "The optional clang-format auto formatter has not been"
-    " found. For more information see"
-    " https://clang.llvm.org/docs/ClangFormat.html")
+    set(ERROR_MESSAGE "clang-format not found, the ${CLANG_FORMAT_TARGET} target have been disabled")
```
Comment (max-veit) [inline cmake/clangformat.cmake:69]:
```suggestion
    set(ERROR_MESSAGE "clang-format not found, so the ${CLANG_FORMAT_TARGET} target is unavailable")
```

File: cmake/clangformat.cmake [hunk 2]
```diff
@@ -62,24 +64,48 @@ mark_as_advanced(CLANG_FORMAT)
...
     OUTPUT_VARIABLE CLANG_FORMAT_VERSION
 )
-string(REGEX REPLACE "clang-format version ([0-9.]+).*" "\\1" CLANG_FORMAT_VERSION "${CLANG_FORMAT_VERSION}")
+string(REGEX REPLACE ".*clang-format version ([0-9.]+).*" "\\1" CLANG_FORMAT_VERSION "${CLANG_FORMAT_VERSION}")
 if("${CLANG_FORMAT_VERSION}" VERSION_LESS "8.0")
-    message(FATAL_ERROR "Unsupported clang-format (version ${CLANG_FORMAT_VERSION}): a newer version (at least 8.0) is required")
+    set(ERROR_MESSAGE
+        "Unsupported clang-format (version ${CLANG_FORMAT_VERSION}): "
+        "a more recent version (at least 8.0) is required."
+        "The ${CLANG_FORMAT_TARGET} target have been disabled."
```
Comment (max-veit) [inline cmake/clangformat.cmake:94]:
```suggestion
        "The ${CLANG_FORMAT_TARGET} target has been disabled."
```

**Review by max-veit — APPROVED** [review]
Great, thanks! Some minor copyediting, but else good to go.

**Pull request merged**